test/unit/org/apache/cassandra/index/sai/v1/StorageAttachedIndexTest.java

```diff
@@ -0,0 +1,166 @@
+        {
+            for (CQLTester.Vector<Float> vectorFloat : vectorFloatList)
+            {
+                QueryProcessor.executeInternal(String.format("INSERT INTO %s.%s (key, value) VALUES (?, ?)", KEYSPACE, TABLE), i, vectorFloat);
+            }
+            cfs.forceBlockingFlush(ColumnFamilyStore.FlushReason.UNIT_TESTS);
+        }
+
+        TableMetadata tableMetadata = cfs.metadata();
+        String columnName = "value";
+        ColumnIdentifier columnIdentifier = new ColumnIdentifier(columnName, true);
+        ColumnMetadata columnDef = tableMetadata.getExistingColumn(columnIdentifier);
+        if (!(columnDef.type instanceof VectorType))
+            throw invalidRequest("ANN is only supported against DENSE FLOAT32 columns");
+
+        // Convert List<CQLTester.Vector<Float>> to List<ByteBuffer>
+        byteBufferList = new ArrayList<>();
+        for (CQLTester.Vector<Float> vector : vectorFloatList) {
+            ByteBuffer buffer = floatVectorToByteBuffer(vector);
+            byteBufferList.add(buffer);
+        }
+
+        Term terms = new Lists.Value(byteBufferList);
+
+        testRestriction = new SingleColumnRestriction.AnnRestriction(columnDef, terms);
+
+        sai = (StorageAttachedIndex) cfs.getIndexManager().getIndexByName(String.format("%s_value_idx", TABLE));
+    }
+
+    @SafeVarargs
+    protected final <T> CQLTester.Vector<T> vector(T... values)
+    {
+        return new CQLTester.Vector<>(values);
+    }
+
+    protected CQLTester.Vector<Float> vector(float[] v)
+    {
+        var v2 = new Float[v.length];
+        for (int i = 0; i < v.length; i++)
+            v2[i] = v[i];
+        return new CQLTester.Vector<>(v2);
+    }
+
+    private static ByteBuffer floatVectorToByteBuffer(CQLTester.Vector<Float> vector) {
+        ByteBuffer buffer = ByteBuffer.allocate(vector.size() * 4);  // 4 bytes per float
+        for (Float value : vector) {
+            buffer.putFloat(value);
+        }
+        buffer.flip();
+        return buffer;
+    }
+
+    @Test
+    public void testOrderResults() {
+        ResultSet.ResultMetadata resultMetadata = new ResultSet.ResultMetadata(new ArrayList<>(cfs.metadata.get().columns()));
+        ResultSet resultSet = new ResultSet(resultMetadata);
+        QueryOptions queryOptions = QueryOptions.create(ConsistencyLevel.ONE,
+                                                        byteBufferList,
+                                                        false,
+                                                        PageSize.inRows(1),
+                                                        null,
+                                                        null,
+                                                        ProtocolVersion.CURRENT,
+                                                        KEYSPACE);
+
+        List<List<ByteBuffer>> rows = new ArrayList<>();
+        rows.add(byteBufferList);
+        resultSet.rows = rows;
```

Comment:
Nit: I was able to get the test to pass with this modification, which also means we can make the `row` field `final`.

```suggestion
        QueryOptions queryOptions = QueryOptions.create(ConsistencyLevel.ONE,
                                                        byteBufferList,
                                                        false,
                                                        PageSize.inRows(1),
                                                        null,
                                                        null,
                                                        ProtocolVersion.CURRENT,
                                                        KEYSPACE);
        List<List<ByteBuffer>> rows = new ArrayList<>();
        rows.add(byteBufferList);
        ResultSet resultSet = new ResultSet(resultMetadata, rows);
```

Reply:
Thanks for this - I agree and addressed it [here](https://github.com/datastax/cassandra/pull/710/commits/fff4a3327f31f2a812bd8bc77472ba423005c860).